ballerina-tests/tests/http_listener_method_service_test.bal

```diff
@@ -47,18 +48,55 @@ service /startService on listenerMethodListener {
                     // gracefulStop will unbind the listener port and stop accepting new connections.
                     // But already connection created clients can communicate until client close.
                     lock {
-                        check listenerMethodbackendEP.gracefulStop();
+                        check listenerMethodGracebackendEP.gracefulStop();
                     }
                     error? responseToCaller = caller->respond("Mock2 invoked!");
                     if responseToCaller is error {
                         log:printError("Error sending response from mock service", 'error = responseToCaller);
                     }
                 }
             };
-            check listenerMethodbackendEP.attach(listenerMethodMock2, "mock2");
+            check listenerMethodGracebackendEP.attach(listenerMethodMock2, "mock2");
         }
         lock {
-            check listenerMethodbackendEP.start();
+            check listenerMethodGracebackendEP.start();
+        }
+        error? result = remoteCaller->respond("Backend service started!");
+        if result is error {
+            log:printError("Error sending response", 'error = result);
+        }
+    }
+    resource function get testImmediate(http:Caller remoteCaller) returns error? {
+        lock {
+            http:Service listenerMethodMock1 = service object {
```

Comment:
We can't do that as `listenerMethodGracebackendEP` is an isolated object which defined globally and excluding `listenerMethodMock1` from the lock leads to the following error
`invalid access of an 'isolated' variable outside a 'lock' statement`

Reply:
Also listenerMethodGracebackendEP is isolated to preserve the isolation root. Otherwise following warnings will be appeared.
`concurrent calls will not be made to this method since the method is not an 'isolated' method`

Reply:
Can't we extract out `listenerMethodMock1` as a global variable and mark it `final` and `readonly` ? Won't that fix the issue ?

Reply:
As discussed, listenerMethodMock1 cannot be final readonly as the attach API expects just the `http:Service`.